### PR TITLE
[IMP] web_editor: improved link editor

### PR DESCRIPTION
**Pull request by pga-odoo**
TASK: https://www.odoo.com/web#id=28341&view_type=form&model=project.task&action=333&active_id=131&menu_id=4720
PAD: https://pad.odoo.com/p/r.f8331ab210ad360b1d17ed2ddef806e6

Description of the issue/feature this PR addresses:
- Added buttons for theme colors alpha, beta, gamma, delta, epsilon
- Added Color selector for choosing button color
- Removed select2 page selector and used jquery autocomplete on same input
- Fixed issue ir_http, getting None in request.website

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website/models/ir_http.py [hunk 1]
```diff
@@ -371,8 +371,8 @@ def generate(self, query=None, args={}):
         query = query and query.startswith('website.') and query[8:] or query
         if query:
             domain += [('key', 'like', query)]
-        website_id = request.context.get('website_id') or request.env['website'].search([], limit=1).id
-        domain += ['|', ('website_id', '=', website_id), ('website_id', '=', False)]
+        website = request.env['website'].get_current_website()
```
Comment (tivisse) [inline addons/website/models/ir_http.py]: Even if it seems right to me. Can you explain me the reason of this modification ? Is it in the purpose of the task ?

File: addons/website/models/website.py [hunk 2]
```diff
@@ -527,7 +527,7 @@ def enumerate_pages(self, query_string=None):
                       of the same.
             :rtype: list({name: str, url: str})
         """
-        request.context = dict(request.context, **self.env.context)
+        # request.context = dict(request.context, **self.env.context)
```
Comment (tivisse) [inline addons/website/models/website.py]: Why do you remove this line ?

File: addons/website/models/ir_http.py (hunk 1 above)
Comment (pga-odoo) [inline addons/website/models/ir_http.py]: Hello @tivisse 
When i was developing this task page search was not working in master. So i have fixed it in my branch. During my development someone make fix for it at https://github.com/odoo/odoo/commit/a98d60cb11ceee31a6bd9d3734d0e4b565e5d677  **which was not working either** so i reverted that commit in branch and fix it again.

Now during that another fix has been made to fix previous broken fix   https://github.com/odoo/odoo/commit/273a15ea7d6e23e14b939c518fd1d7cfcd0cff94 

should i remove it or keep it ???

File: addons/website/models/website.py (hunk 2 above)
Comment (pga-odoo) [inline addons/website/models/website.py]: it was from same broken fix https://github.com/odoo/odoo/commit/a98d60cb11ceee31a6bd9d3734d0e4b565e5d677

File: addons/website/models/ir_http.py (hunk 1 above)
Comment (tivisse) [inline addons/website/models/ir_http.py]: Thanks for the answer ! You can keep it, it seems good to me.

Comment (pga-odoo) [inline addons/website/models/ir_http.py]: :+1:

**Pull request merged**